### PR TITLE
Completed week 2 assessment

### PR DESCRIPTION
**Pull request by iraholmes**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [✅  ] Coding Challenges
- [ ✅ ] Coding Concepts
- [✅  ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:
I would like to get more information about what I was missing on the final jest question, I felt that I got so close then got lost. 
Also on the last question in the coding concepts section. Thank you

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -23,32 +25,100 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
+    if (divide.number % 3 === 0){
+        return `${divide.number} is divisible by three`
+    } else if (divide.number % 3 === 0){
+        return `${divide.numer} is divisble by three`
+    } else if (divide.number % 3 !== 0){
+        return `${divide.number} is not divisible by three`
+    } else {
+        return "Math not mathing"
+    }
+}
```
Comment (Genemichael25) [inline code-challenges.test.js:60]: Great job on this function! You don't need to check if the number % 3 === 0 twice, I would get rid of the else if and return on line 55-56. Other than that, looks good!

File: code-challenges.test.js [hunk 2]
```diff
@@ -23,32 +25,100 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
+// Snapshots:   0 total
+// Time:        0.852 s, estimated 1 s
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
+//Psuedo: Create a function called capNoun that takes in the array of words.
+// Use built in method .toString to change the Array into a string.
+// connect allCaps variable to both arrays
+//Return: An array with all capitol characters 
+
```
Comment (Genemichael25) [inline code-challenges.test.js:72]: The intent with this challenge is to capitalize the first letter of the strings in an array. What do you mean by connect allCaps variable to both arrays?

File: code-challenges.test.js [hunk 3]
```diff
@@ -23,32 +25,100 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
-// --------------------3) Create a function that takes in a string and logs the index of the first vowel.
-
+const capNoun = (allCaps) => {
+    if(allCaps.randomNouns1 == "string"){
+        return (["STREETLAMP", "POTATO", "TEETH", "CONCLUSION", "NEPHEW"])
+    } else if (allCaps.randomNouns2 == "string"){
+        return (["Temperature", "DATABASE", "CHOPSTICKS", "MANGO"])
+    }
+}  
+// Test Suites: 1 passed, 1 total
```
Comment (Genemichael25) [inline code-challenges.test.js:100]: Do we need randomNouns in our function?

File: code-challenges.test.js [hunk 4]
```diff
@@ -23,32 +25,100 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
+// Tests:       2 passed, 2 total
+// Snapshots:   0 total
+// Time:        0.4 s, estimated 1 s
+
+        // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
+//Psuedo: Create a function called firstVowel
+//
```
Comment (Genemichael25) [inline code-challenges.test.js]: This pseudocode isn't enough to start visualizing how we can create the code for this problem.

File: coding-concepts.js [hunk 5]
```diff
@@ -10,7 +10,7 @@
 const cohort = "Alpha 2023"
 // console.log(cohort.split(""))
 
-// a) Your answer:
+// a) Your answer: Each letter in the string will be separated into their own string. "A" "l""p""h""a" "2" "0""2""3"
 // b) Verify and explain:
```
Comment (Genemichael25) [inline coding-concepts.js]: Missing verification and explanation.

File: coding-concepts.js [hunk 6]
```diff
@@ -20,24 +20,25 @@ const greeter = (name) => {
...
 
 // --------------------3) What will this log?
 
 const multipliedByTwo = [4, 5, 6, 7, 8].map((number) => number * 2)
 // console.log(multipliedByTwo)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: Each number in the array will be multiplied by 2 [8,10,12,14,16]
+// b) Verify and explain:[ 8, 10, 12, 14, 16 ] An array of all numbers multiplied by 2, because the .map() method ran each array number through the * 2 action and returned them.
```
Comment (Genemichael25) [inline coding-concepts.js:33]: Exactly, great job explaining this one!

File: coding-concepts.js [hunk 7]
```diff
@@ -47,7 +48,7 @@ const myCodingSkills = {
   databases: "PostgreSQL",
   versionControl: "GitHub"
 }
-// console.log(myCodingSkills.languages[0])
+console.log(myCodingSkills.languages[0])
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: This will return the 0 index for each object's key [J],[R][P][G]
+// b) Verify and explain: JavaScript, the entire first word in the string was returned. The string at the 0 index was returned.
```
Comment (Genemichael25) [inline coding-concepts.js]: Why was just JavaScript returned?

File: interview-questions.md [hunk 8]
```diff
@@ -8,50 +8,56 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
+while .filter() returns the output of a conditional statement narrowing down to more specific values such as returning the value at index 3.
 
 1. What is iteration?
 
-Your answer:
+Your answer: The constant running of a function by the terminal based on parameters given to return a desire outcome.
 
-Researched answer:
+Researched answer: Iteration is the protocol that runs the code to the next operator to return a value or true / false response.
```
Comment (Genemichael25) [inline interview-questions.md:37]: Iteration is performing something on each element in a collection. For example, with .map we can multiply the values in an array by a number. Going into each element is the iteration.

**Review by Genemichael25 — CHANGES_REQUESTED** [review]
Ira, I am requesting changes for the following reasons: code challenges #2 test/code needs to be refactored for the question, #3 not sufficient pseudocode. Coding concept #1 needs verification and explanation. Please let us know if you don't understand a question, or need clarification on the feedback I have left throughout.

File: code-challenges.test.js [hunk 9]
```diff
@@ -23,32 +25,113 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
+// Use built in method .toString to change the Array into a string.
+// connect allCaps variable to both arrays
+//Return: An array with the first letter of each word capitolized  
+
 // a) Create a test with expect statements for each of the variables provided.
 
+
+describe("capNoun", () => {
+    it("returns the first letter of each word in an array capitalized", () => {
+      expect(capNoun(randomNouns1[0])).toString("Streetlamp", "Potato", "Teeth", "Conclusions", "Nephew")
```
Comment (elysemontano) [inline code-challenges.test.js:78]: Keep in mind your test should not have any logic.  We want the test to simply invoke the function with any relevant arguments and check that the outcome equals an expected output.

**Review by elysemontano — APPROVED** [review]
Thank you for making the changes Ira!  Keep working on setting up the test so that it is invoking the function with any arguments and comparing against a set expected output.  Also make sure to keep working on breaking down the problems into small pieces in your pseudocode. This will really help bring the logic together.  Please make sure to reach out if you need extra help.  We have some really great mentors that we can set you up with to give you some extra assistance.

**Pull request merged**